runtime/jcl/common/mgmtthread.c

```diff
@@ -1257,7 +1248,7 @@ initIDCache(JNIEnv *env)
 	if (JNI_ENOMEM == err) {
 		(*env)->ThrowNew(env, oom, "initIDCache failed");
 	}
-	return err;
+	throwError(currentThread, err);
```

Comment:
I think `else` is missing here (we don't want to replace the exception registered on line 1249).
Also, `throwError()` appears to assume we've called `internalEnterVMFromJNI()` which is not the case here.

Reply:
Agreed - failing JNI calls almost always set a pending exception so we should propagate those. There' should be no need to track a JNI error here.

Reply:
so what I understand is that we don't need an explicit throw, as the JNI - to Java transition has mechanisms to handle that. So I removed existing logic that propagates errors (other than allocation failures). let me know if that is what the essence of both the comments.